ql/src/java/org/apache/hadoop/hive/ql/exec/repl/ranger/RangerRestClientImpl.java

```diff
@@ -0,0 +1,356 @@
+                                          String targetDbName) {
+    if (sourceDbName.endsWith("/")) {
+      sourceDbName = StringUtils.removePattern(sourceDbName, "/+$");
+    }
+    if (targetDbName.endsWith("/")) {
+      targetDbName = StringUtils.removePattern(targetDbName, "/+$");
+    }
+    if (targetDbName.equals(sourceDbName)) {
+      return rangerPolicies;
+    }
+    if (CollectionUtils.isNotEmpty(rangerPolicies)) {
+      Map<String, RangerPolicy.RangerPolicyResource> rangerPolicyResourceMap = null;
+      RangerPolicy.RangerPolicyResource rangerPolicyResource = null;
+      List<String> resourceNameList = null;
+      for (RangerPolicy rangerPolicy : rangerPolicies) {
+        if (rangerPolicy != null) {
+          rangerPolicyResourceMap = rangerPolicy.getResources();
+          if (rangerPolicyResourceMap != null) {
+            rangerPolicyResource = rangerPolicyResourceMap.get("database");
+            if (rangerPolicyResource != null) {
+              resourceNameList = rangerPolicyResource.getValues();
+              if (CollectionUtils.isNotEmpty(resourceNameList)) {
+                for (int i = 0; i < resourceNameList.size(); i++) {
+                  String resourceName = resourceNameList.get(i);
+                  if (resourceName.equals(sourceDbName)) {
+                    resourceNameList.set(i, targetDbName);
+                  }
+                }
+              }
+            }
+          }
+        }
+      }
+    }
+    return rangerPolicies;
+  }
+
+  private Path writeExportedRangerPoliciesToJsonFile(String jsonString, String fileName, Path stagingDirPath)
+      throws Exception {
+    String filePath = "";
+    Path newPath = null;
+    FSDataOutputStream outStream = null;
+    OutputStreamWriter writer = null;
+    try {
+      if (!StringUtils.isEmpty(jsonString)) {
+        FileSystem fileSystem = getFileSystem(stagingDirPath);
+        if (fileSystem != null) {
+          if (!fileSystem.exists(stagingDirPath)) {
+            fileSystem.mkdirs(stagingDirPath);
+          }
+          newPath = stagingDirPath.suffix(File.separator + fileName);
+          outStream = fileSystem.create(newPath, true);
+          writer = new OutputStreamWriter(outStream, "UTF-8");
+          writer.write(jsonString);
+        }
+      }
+    } catch (IOException ex) {
+      if (newPath != null) {
+        filePath = newPath.toString();
+      }
+      throw new Exception("Failed to write json string to file:" + filePath, ex);
+    } catch (Exception ex) {
+      if (newPath != null) {
+        filePath = newPath.toString();
+      }
+      throw new Exception("Failed to write json string to file:" + filePath, ex);
+    } finally {
+      try {
+        if (writer != null) {
+          writer.close();
+        }
+        if (outStream != null) {
+          outStream.close();
+        }
+      } catch (Exception ex) {
+        LOG.error("Unable to close writer/outStream.", ex);
```

Comment:
In other places, for a failure during stream close, we are throwing exception back. Shouldn't we be consistent in this case as well?